overlord/hookstate/hooks.go

```diff
@@ -375,4 +371,7 @@ func setupHooks(hookMgr *HookManager) {
 	hookMgr.Register(regexp.MustCompile("^pre-refresh$"), handlerGenerator)
 	hookMgr.Register(regexp.MustCompile("^remove$"), handlerGenerator)
 	hookMgr.Register(regexp.MustCompile("^gate-auto-refresh$"), gateAutoRefreshHandlerGenerator)
+	hookMgr.Register(regexp.MustCompile("^change-view-.+$"), ChangeViewHandlerGenerator)
+	hookMgr.Register(regexp.MustCompile("^save-view-.+$"), SaveViewHandlerGenerator)
+	hookMgr.Register(regexp.MustCompile("^.+-view-changed$"), handlerGenerator)
```

Comment:
if you see how this is done by ifacestate for example, the preferred way to deal with this is actually to have the Manager constructor take the hookMgr and do the register